### DISCUSSION TITLE
"supabase db remote commit" does not work like documentation

### DISCUSSION THREAD

**Discussion by sangdth** [supabase/supabase#6907] [Questions] [answered]
Please help!

I'm trying to follow the Local Development guide, https://supabase.com/docs/guides/local-development#linking-your-project

But on the step `supabase db remote commit`, I got this error:
```
Error: The remote database's migration history is not in sync with the contents of supabase/migrations. Resolve this by:
- Updating the project from version control to get the latest supabase/migrations,
- Pushing unapplied migrations with supabase db push,
- Or failing that, manually inserting/deleting rows from the supabase_migrations.schema_migrations table on the remote database.
```

I did try to use `supabase db push`, then running `supabase db remote commit` again, this time it does not show the Error, but stuck here forever:
<img width="576" alt="Screenshot 2022-05-21 at 0 21 43" src="https://user-images.githubusercontent.com/1083478/169620391-c8bb3fcf-2308-400b-95d2-c4e8421831da.png">

I didn't know how to proceed, tried to Google a lot as well as digging the issues, but had no luck.
Thus I deleted my project, created new one, and started from scratch. Currently I'm stuck at the Error.

Thanks.

**Comment by sangdth** [comment, marked as answer]
I tried it again today, running `supabase db push`, then running `supabase db remote commit` again and it goes through. I have no idea why.

IMHO the documentation should add the `supabase db push` before the remote commit.

**Reply by ozozozd** [reply]
Also stuck at `supabase db remote commit` and `supabase db push` also fails complaining with the following:
`Error: supabase_migrations.schema_migrations table conflicts with the contents of supabase/migrations.; Found 1 versions and 0 migrations.`

**Reply by kobago** [reply]
I got this error too. But by removing the conflicting migration version row from the remote Supabase's `supabase_migrations` schema, the error disappeared. Give it a try :)

**Comment by colestriler** [comment]
You can only run `supabase db remote commit` once. I wrote an explanation in a different thread: https://github.com/supabase/supabase/discussions/6366#discussioncomment-4865266

**Comment by tmcw** [comment]
This doesn't seem like a great answer - shouldn't there a ticket cut so that

- Documentation says you can only run this once
- The tool itself doesn't mysteriously stall when the user takes this typical action of running it more than once?
